environment.go

```diff
@@ -164,6 +150,37 @@ nextExt:
 		default:
 		}
 	}
+	// This is here as we could've not run go mod tidy due to a replace being the only extension
+	err = env.execGoModTidy(ctx)
```

Comment:
THe for loop above that adds the extensions completely *skips* `go mod *` commands if it was also `replaced` aka `--with someextension=.` (as an example). But still writes the `.go` file that will require it. 

I am not certain if we can remove that check though as ... it will just do nothing AFAIK :thinking: And call `go mod tidy` just as well. But I guess then I will need comment there why we are not skipping the ones that are replaced :facepalm:

Reply:
unfortunately if we haven't run `go mod tidy` in between `go mod edit -require`s which don't give exact version but instead use stuff such as `latest` it will error out, which basically fails the build. So for this reason I prefer to call `go mod tidy` 2 more times than anything else :). 

I did wonder if i should have some boolean on `env` to say if `go mod tidy` was the last command ran and if yes to skip running it, but it seems that they do run *very* quickly when there is nothing to do so it seems not needed IMO